hazelcast/src/main/java/com/hazelcast/nio/ascii/TextChannelInboundHandler.java

```diff
@@ -214,13 +217,15 @@ private boolean isCommandTypeEnabled(TextCommand command) {
                 boolean isHealthCheck = healthcheckEnabled && uri.startsWith(HttpCommandProcessor.URI_HEALTH_URL);
                 boolean forceRequestHandling = isClusterManagementRequest || isMancenterRequest || isHealthCheck;
                 if (!restEnabled && !forceRequestHandling) {
-                    connection.close("REST not enabled", null);
+                    String msg = sslEnabled ? "REST not enabled" : "REST or SSL not enabled";
```

Comment:
what if the SSL is enabled on a member side, but not on a client side?

Reply:
You will get a handshaking error.